docs: documentation overhaul

takb opened the pull request:
### Information about the changes
Reason for change:

- restructured for better overview
- includes YAML configuration 
- contributor docs still WIP

takb committed:
docs(getting-started): finishing up structure rework

Co-authored-by: Jakob Schnell <Jakob.Schnell@heigit.org>
Co-authored-by: sfendrich <sascha.fendrich@uni-heidelberg.de>
Co-authored-by: Amandus Butzer <amandus.butzer@heigit.org>
Co-authored-by: Jochen Haeussler <jochen.haeussler@heigit.org>
Co-authored-by: Andrzej K. Oleś <andrzej.oles@gmail.com>

MichaelsJP commented on docs/run-instance/installation/running-jar-war.md:
```diff
@@ -0,0 +1,70 @@
+
+[//]: # (TODO: overhaul contents below after integrating the jar build PR)
+
+# Installing and running tomcat9
```
Tomcat 10, no?

takb replied:
sure, but like the comment says this document is WIP and needs overhaul after we finally integrate the JAR build feature...

MichaelsJP replied:
Relating the Jar yes. But if someone opens this doc and tries to start ors with latest master and Tomcat 9, it'll fail due to the latest spring update. Am I mixing something up?

MichaelsJP replied:
Tomcat 10, on the other hand, should be downwards compatible

takb replied:
Tomcat 10 has only packages for ubuntu 23 which is not a LTS release. We also want to move to recommend running the jar and only provide war for users who already have a tomcat set up, because we think that's the only scenario where anyone would insist on using the war. And then we shouldn't try and maintain docs for tomcat that also would be strongly platform dependent.
I'd rather remove that section entirely and add it back after merging the jar thing and making sure what we want to have in the docs regarding the war at all...

takb replied:
I've updated the file to contain no wrong information.

MichaelsJP commented on docs/run-instance/installation/running-jar-war.md:
```diff
@@ -0,0 +1,70 @@
+# Running JAR / WAR
```
The runnin as JAR part is not in the actual docs

takb replied:
so is the code

MichaelsJP commented on ors-engine/src/main/java/org/heigit/ors/routing/configuration/RouteProfileConfiguration.java:
```diff
@@ -64,7 +64,7 @@ public class RouteProfileConfiguration {
     private int maximumVisitedNodesPT = 1000000;
 
-    private boolean turnCostEnabled = false;
+    private boolean turnCostEnabled = false;//FIXME: even though the field is read by external methods, its setter is never called.
```
Shouldn't be the FIXME in code instead of the docs?

takb replied:
It is, this appears to be an actual smell in the code @aoles came across while writing the docs (I'm guessing), that's why the comment got mixed in here with the docs commits. I personally think we shouldn't get too finicky about the commit/PR semantics...

aoles replied:
That's correct, I figured out we might want to change it. Obviously addressing the TODO is outside of the PR's scope .

MichaelsJP commented on ors-engine/src/main/java/org/heigit/ors/routing/graphhopper/extensions/ORSGraphHopper.java:
```diff
@@ -88,8 +88,8 @@ public class ORSGraphHopper extends GraphHopperGtfs {
-    private int minNetworkSize = 200;
-    private int minOneWayNetworkSize = 0;
+    private int minNetworkSize = 200;//TODO: does this parameter actually influence anything?
+    private int minOneWayNetworkSize = 0;//TODO: does this parameter actually influence anything?
```
Same here. What to do with a TODO tag in the docs?
Probably not the best location for it.

takb replied:
same as above

MichaelsJP commented on docker-compose.yml:
```diff
@@ -19,6 +19,7 @@ services:
       - ./docker/logs/tomcat:/home/ors/tomcat/logs
       - ./docker/conf:/home/ors/ors-conf
       - ./docker/data:/home/ors/ors-core/data
+      - ../mapdata/hoyerswerda.pbf:/home/ors/ors-core/data/osm_file.pbf
```
What? 😄

takb replied:
oops

MichaelsJP reviewed (changes requested):
Just a coupble of things while looking over a few minutes

koebi commented on docs/.vitepress/config.js:
```diff
@@ -34,81 +34,108 @@ export default defineConfig({
+                                    {text: 'Isochrones', link: '/api-reference/endpoints/isochrones/'},
+                                    {text: 'Matrix', link: '/api-reference/endpoints/matrix/'},
+                                    {text: 'Snapping', link: '/api-reference/endpoints/snapping/'},
+                                    {text: 'Export', link: '/api-reference/endpoints/export/'},
```
Snapping and Export are both not live endpoints, no?
We're currently releasing the docs anytime they get pushed to master, so maybe we want to comment this out until they're live?

jhaeu replied:
Maybe we could leave them in the doc (for people running the ORS themselfes), but clearly mark them as not part of our live API? In the side bar and on the pages themselfes?

koebi replied:
Yes, sounds good :)
To me, a clearly marked comment on the page itself (e.g. like the blue info boxes in confluence) should suffice, but we could also mention sth in the sidebar.

jhaeu replied:
I like the idea with the info boxes - changed it accordingly, thanks!

sfendrich replied:
I think we want to activate snapping on our live API.
The export endpoint should be marked as experimental, as it will probably change a lot in the future.

koebi commented on docs/getting-started.md:
```diff
@@ -1,20 +1,25 @@
+* [Run ORS instance](run-instance/) contains all info about setting up the ors locally. 
+  * [System requirements](run-instance/system-requirements)
+  * [Data](run-instance/data) used by openrouteservice
+  * [Installation](run-instance/installation/) either directly using [JAR / WAR](run-instance/installation/running-jar-war) artifacts, with [Docker](run-instance/installation/running-in-container) or [building from source](run-instance/installation/building-from-source)
```
Link to docker is wrong:
```suggestion
  * [Installation](run-instance/installation/) either directly using [JAR / WAR](run-instance/installation/running-jar-war) artifacts, with [Docker](run-instance/installation/running-with-docker) or [building from source](run-instance/installation/building-from-source)
```

koebi commented on docs/technical-details/index.md:
```diff
@@ -0,0 +1,7 @@
+[//]: # (Attention: Keep docu up-to-date!)
+[//]: # (Alternative: Use Github-Code-Links to enums etc. in documentation?)
+
+See some value lists in sub pages
```
There could be a bit more information here, e.g. a list of the sub-pages w/ a small overview as with the getting started page

jhaeu replied:
added overview

koebi commented on docs/api-reference/index.md:
```diff
@@ -0,0 +1,28 @@
+
+There are different ways to get information about the openrouteservice API:
+
+* **First starting point:** [API interactive examples](https://openrouteservice.org/dev/#/api-docs/directions%20service): 
```
```suggestion
* **First starting point:** [API interactive examples](https://openrouteservice.org/dev/#/api-docs/directions_service): 
```

koebi commented on docs/api-reference/index.md:
```diff
@@ -0,0 +1,28 @@
+  Explanation and examples for some selected topics that might not be explained sufficiently in the interactive examples. 
+  Here, the single endpoints are not completely covered with all their request parameters etc. 
+  But on the other hand, there is some information about endpoints that are not available in our live API, 
+  but e.g. on instances you run or host yourself: [Export](endpoints/export/index.md), [Health](endpoints/health/index.md), [Status](endpoints/status/index.md)
```
Snapping is missing, note comment on config.js

koebi commented on docs/api-reference/endpoints/index.md:
```diff
@@ -0,0 +1,27 @@
+* [Directions Service](directions/index.md): Get directions for different modes of transport
+* [Isochrones Service](isochrones/index.md): Obtain areas of reachability from given locations
+* [Matrix Service](matrix/index.md): Obtain one-to-many, many-to-one and many-to-many matrices for time and distance
+* [Snapping Service](snapping/index.md): Snap coordinates to the graph edges
```
Same as in other notes on Snapping

koebi commented on docs/api-reference/endpoints/index.md:
```diff
@@ -0,0 +1,27 @@
+* [POI Service](poi-service): Stand-Alone service from HeiGIT that returns points of interest in the area surrounding a geometry
+* [Elevation Service](https://openrouteservice.org/dev/#/api-docs/elevation): Stand-Alone service from HeiGIT that returns the elevation for point or line geometries 
+* [Geocoder](geocoder/index.md): A Third Party Service ([Pelias](https://www.pelias.io)) that resolves geo coordinates to addresses and vice versa
+* [Optimization](https://openrouteservice.org/dev/#/api-docs/optimization): Optimize routes for vehicle fleets (-> [vroom](https://github.com/VROOM-Project/vroom))
```
Two of these link internally, two externally. I suggest setting up a page for Elevation and Optimization as well.
As for the comment/link to vroom, I think that could need a bit more info - a mention here is fine (as with pelias) and a bit more can go on the own sub-page.

jhaeu replied:
Good points. I tried to write something, I hope, it's enough? Please check again @koebi

koebi commented on docs/api-reference/endpoints/directions/index.md:
```diff
@@ -0,0 +1,12 @@
+
+The Directions Service calculates routes and directions between two or more locations for different modes of transportation. 
+
+A complete and concise overview of all request parameters and return types can be found in the [interactive API documentation](https://openrouteservice.org/dev/#/api-docs/directions%20service).
```
```suggestion
A complete and concise overview of all request parameters and return types can be found in the [interactive API documentation](https://openrouteservice.org/dev/#/api-docs/directions_service).
```

koebi commented on docs/api-reference/endpoints/directions/requests-and-return-types.md:
```diff
@@ -0,0 +1,33 @@
+**GeoJSON** is a format for encoding a variety of geographic data structures, see [geojson.org](https://geojson.org).
+It is widely used and can therefore be easily processed or displayed in many applications, e.g. in [QGIS](https://qgis.org/) or on [geojson.io](http://geojson.io/)
+
+More information about the result types can be found in the [interactive API documentation](https://openrouteservice.org/dev/#/api-docs/directions%20service).
```
```suggestion
More information about the result types can be found in the [interactive API documentation](https://openrouteservice.org/dev/#/api-docs/directions_service).
```

koebi commented on docs/api-reference/endpoints/directions/routing-options.md:
```diff
@@ -0,0 +1,273 @@
+# Routing Options
```
This is great, we should probably link to this from the API Playground? @TheGreatRefrigerator

koebi commented on docs/api-reference/endpoints/directions/routing-options.md:
```diff
@@ -0,0 +1,273 @@
+| `steps`    | cycling-\*, foot-\*, wheelchair             |
+
+### `avoid_polygons` 
+Comprises areas to be avoided for the route. Formatted as [geojson polygon](http://geojson.org/geojson-spec.html#id4) or [geojson multipolygon](http://geojson.org/geojson-spec.html#id7).
```
We should link to RFC 7946 here.

```suggestion
Comprises areas to be avoided for the route. Formatted as [geojson polygon](https://datatracker.ietf.org/doc/html/rfc7946#appendix-A.3) or [geojson multipolygon](https://datatracker.ietf.org/doc/html/rfc7946#appendix-A.6).
```

koebi commented on docs/api-reference/endpoints/directions/requests-and-return-types.md:
```diff
@@ -0,0 +1,33 @@
+
+### GeoJSON
+
+**GeoJSON** is a format for encoding a variety of geographic data structures, see [geojson.org](https://geojson.org).
```
We should link to RFC 7946 here, since geojson.org is obsolete?
```suggestion
**GeoJSON** is a format for encoding a variety of geographic data structures, see [geojson.org](https://datatracker.ietf.org/doc/html/rfc7946).
```

koebi commented on docs/api-reference/endpoints/directions/routing-options.md:
```diff
@@ -0,0 +1,273 @@
+  | Parameter             | Type    | Description                                                                                                                 |
+  |-----------------------|---------|-----------------------------------------------------------------------------------------------------------------------------|
+  | `surface_type`        | String  | Specifies the minimum [surface type](http://wiki.openstreetmap.org/wiki/Key:surface). Default is `"cobblestone:flattened"`. |
+  | `track_type`          | String  | Specifies the minimum [grade](http://wiki.openstreetmap.org/wiki/Key:tracktype) of the route. Default is `"grade1"`.        |
```
Especially since we're using `gradient` above, people may confuse this with the incline.
```suggestion
  | `track_type`          | String  | Specifies the minimum [quality](http://wiki.openstreetmap.org/wiki/Key:tracktype) of the route. Default is `"grade1"`.        |
```
Maybe something like *surface firmness* might even be better, the OSM wiki may provide inspiration ;)

koebi commented on docs/api-reference/endpoints/directions/routing-options.md:
```diff
@@ -0,0 +1,273 @@
+
+### `round-trip` 
+
+[//]: # (TODO: write)
```
As it says, ToDo :D

jhaeu replied:
added something, @koebi

koebi commented on docs/api-reference/endpoints/directions/routing-options.md:
```diff
@@ -0,0 +1,273 @@
+### `vehicle_type`
+For `profile=driving-hgv` only. It is needed for **vehicle restrictions** to work. Possible values:
+
+* `hgv`
+* `bus`
+* `agricultural`
+* `delivery`
+* `forestry` 
+* `goods`
```
This could use info on what's what, or links to the OSM wiki?

TheGreatRefrigerator replied:
@koebi I've linked to the tag filtering instead, where it is described to which osm keys the values are connected

koebi commented on docs/api-reference/endpoints/directions/routing-options.md:
```diff
@@ -0,0 +1,273 @@
+
+Some options examples in readable and minified JSON form:
+
+for `profile=driving-car`:
```
```suggestion
### for `profile=driving-car`:
```

koebi commented on docs/api-reference/endpoints/directions/routing-options.md:
````diff
@@ -0,0 +1,273 @@
+```
+`{"avoid_features":["ferries","tollways"]}`
+
+for `profile=cycling-*`:
````
```suggestion
### for `profile=cycling-*`:
```

koebi commented on docs/api-reference/endpoints/directions/routing-options.md:
````diff
@@ -0,0 +1,273 @@
+```
+`{"avoid_features":["steps"],"profile_params":{"weightings":{"steepness_difficulty":2}}}},"avoid_polygons":{"type":"Polygon","coordinates":[[[100.0,0.0],[101.0,0.0],[101.0,1.0],[100.0,1.0],[100.0,0.0]]]}}`
+
+for `profile=foot-*`:
````
```suggestion
### for `profile=foot-*`:
```

koebi commented on docs/api-reference/endpoints/directions/routing-options.md:
````diff
@@ -0,0 +1,273 @@
+```
+`{"avoid_features":["fords","ferries"],"profile_params":{"weightings":{"green":{"factor":0.8},"quiet":{"factor":1.0}}},"avoid_polygons":{"type":"Polygon","coordinates":[[[100.0,0.0],[101.0,0.0],[101.0,1.0],[100.0,1.0],[100.0,0.0]]]}}`
+
+for `profile=driving-hgv`:
````
```suggestion
### for `profile=driving-hgv`:
```

koebi commented on docs/api-reference/endpoints/directions/routing-options.md:
````diff
@@ -0,0 +1,273 @@
+```
+`{"avoid_features":["ferries","tollways"],"vehicle_type":"hgv","profile_params":{"restrictions":{"length":30,"width":30,"height":3,"axleload":4,"weight":3,"hazmat":true}},"avoid_polygons":{"type":"Polygon","coordinates":[[[100.0,0.0],[101.0,0.0],[101.0,1.0],[100.0,1.0],[100.0,0.0]]]}}`
+
+for `profile=wheelchair`:
````
```suggestion
### for `profile=wheelchair`:
```

koebi commented on docs/api-reference/endpoints/directions/routing-options.md:
```diff
@@ -0,0 +1,273 @@
+
+
+
+## Examples
```
We should probably check the examples to make sure they work.

TheGreatRefrigerator replied:
Done

koebi commented on docs/api-reference/endpoints/directions/extra-info/index.md:
```diff
@@ -0,0 +1,149 @@
+
+When requesting routes, there are a number of "extra info" items that can be requested to give you more information about the route. 
+This info could be things like the road surface, track type, or OpenStreetMap way ID. 
+The list below details which extra info items are available for each profile in the routing provided by https://api.openrouteservice.org.
```
I feel like this page should link to the backend documentation as well. What do you think @TheGreatRefrigerator

TheGreatRefrigerator replied:
Done

koebi commented on docs/api-reference/endpoints/directions/extra-info/index.md:
```diff
@@ -0,0 +1,149 @@
+|           waytype            | Types of roads and paths that are used in the route                                                                                          | `waytypes`¹              | [Type IDs](waytype.md)                                       |
+|           tollways           | Any tollways that the route crosses. Whether a way is marked as [tollway](https://wiki.openstreetmap.org/wiki/Key:toll) depends on `profile` | `tollways`               | 0 (no tollway) or 1 (tollway)                                |
+|       traildifficulty        | The difficulty of parts of the way based on sac or mountainbike scales                                                                       | `traildifficulty`        | [Difficulty IDs](trail-difficulty.md)                        |
+|            osmid             | The OpenStreetMap way IDs of the ways the route uses                                                                                         | `osmId`¹                 |                                                              |
```
Should there be an OSM-Wiki-Link here?

TheGreatRefrigerator replied:
Done

koebi commented on docs/api-reference/endpoints/directions/extra-info/index.md:
```diff
@@ -0,0 +1,149 @@
+|            green             | How "green" the parts of the route are (influenced by things like number of trees, parks, rivers etc.)                                       | `green`                  | 0 (minimal greenspace) - 10 (a lot of green space)           |
+|            noise             | How noisy the parts of the route are (influenced by things like proximity to highways)                                                       | `noise`                  | 0 (quiet) - 10 (noisy)                                       |
+|            shadow            |                                                                                                                                              | `shadow`                 |                                                              |
+|             csv              |                                                                                                                                              | `csv`                    |                                                              |
```
There's information missing for both of these. We should provide it or take it out.

jhaeu replied:
wrote something, @koebi please resolve if OK

koebi commented on docs/api-reference/endpoints/directions/extra-info/steepness.md:
```diff
@@ -15,3 +21,5 @@ Negative values indicate decline, positive incline.
 | 4     | 12-15%   |
 | 5     | >16%     |
+
+[//]: # (keep in sync with )
```
…with what?

Also, should we have a `-` before the encoding values of negatives?

koebi commented on docs/api-reference/endpoints/directions/extra-info/waycategory.md:
```diff
@@ -13,3 +19,5 @@ The exponential assignment of the values is used for [bit fields](http://eddmann
 | 64    | Paved road                   |
 | 128   | Ford                         |
+
+[//]: # (keep in sync with )
```
…with what?

Can we also link to the OSM Wiki here somewhere?

koebi commented on docs/api-reference/endpoints/directions/extra-info/waytype.md:
````diff
@@ -1,5 +1,11 @@
+```jsonpath
+$.routes[*].extras.waytype.values
+```
+
````
Can we link to the OSM Wiki here somewhere?

TheGreatRefrigerator replied:
this should rather be linked to the ors code on which osm things are used here, as this again are not the exact values of a single osm tag

TheGreatRefrigerator replied:
Nice table

koebi commented on docs/api-reference/endpoints/directions/extra-info/road-access-restrictions.md:
````diff
@@ -0,0 +1,22 @@
+```
+
+Provides information about possible restrictions on roads.
+Explanation of the values can be found in the [list of possible values](https://wiki.openstreetmap.org/wiki/Key:access)
````
```suggestion
Explanation of the values can be found in the [list of possible values in the OSM Wiki](https://wiki.openstreetmap.org/wiki/Key:access)
```

koebi commented on docs/technical-details/country-list.md:
```diff
@@ -1,6 +1,6 @@
 # Country List
```
Should we have this as a single very long list?
People aren't used to scrolling sideways…

takb replied:
done

koebi commented on docs/api-reference/endpoints/isochrones/index.md:
````diff
@@ -0,0 +1,14 @@
+$.features
+```
+
+For an overview of all features of the isochrones endpoint please refer to [interactive API documentation](https://openrouteservice.org/dev/#/api-docs/directions%20service).
````
```suggestion
For an overview of all features of the isochrones endpoint please refer to [interactive API documentation](https://openrouteservice.org/dev/#/api-docs/isochrones_service).
```

koebi commented on docs/api-reference/endpoints/matrix/index.md:
```diff
@@ -0,0 +1,60 @@
+
+The Matrix Endpoint calculates profile specific distance and time matrices between multiple source and destination positions.
+
+Besides parameters for the routing profile, the desired metrics (distance or time) and some more (see  [interactive API-Doc](https://openrouteservice.org/dev/#/api-docs/matrix%20service)), 
```
```suggestion
Besides parameters for the routing profile, the desired metrics (distance or time) and some more (see  [interactive API-Doc](https://openrouteservice.org/dev/#/api-docs/matrix_service)), 
```

koebi commented on docs/api-reference/endpoints/matrix/index.md:
```diff
@@ -0,0 +1,60 @@
+
+To specify whether distances or durations (or both) are to be calculated, the `metrics` parameter can be set accordingly.
+
+For details about all request parameters and the response type, see the [interactive API-Doc](https://openrouteservice.org/dev/#/api-docs/matrix%20service).
```
```suggestion
For details about all request parameters and the response type, see the [interactive API-Doc](https://openrouteservice.org/dev/#/api-docs/matrix_service).
```

koebi commented on docs/api-reference/endpoints/snapping/index.md:
```diff
@@ -0,0 +1,80 @@
+# Snapping Endpoint
+
```
Should we add an attention note of some kind here to make people alert that this endpoint is currently not available in the live API?

koebi commented on docs/api-reference/endpoints/export/index.md:
```diff
@@ -0,0 +1,456 @@
+In the request, the desired routing profile is specified as the last path parameter, 
+a bounding box for the area of interest has to be defined in the request body.
+
+The following example request and response should be pretty self-explanatory:
```
…but a comment about the `weight` in the return value is probably advisable :)

koebi commented on docs/api-reference/endpoints/status/index.md:
```diff
@@ -0,0 +1,99 @@
+
+The status endpoint os not available in the public API.
+
+The GET request http://localhost:8082/ors/v2/status (host and port might be different) returns basic information about the running instance:
```
```suggestion
The GET request http://localhost:8082/ors/v2/status (host and port are dependent on the setup) returns basic information about the running instance:
```

koebi commented on docs/api-reference/endpoints/geocoder/index.md:
```diff
@@ -0,0 +1,9 @@
+The Geocoder Endpoint of our public API is served by a [Pelias](https://www.pelias.io) instance.
+It resolves geo coordinates to addresses and vice versa.
+
+Watchout our [API interactive examples](https://openrouteservice.org/dev/#/api-docs/geocode) to see 
```
```suggestion
Check our [API interactive examples](https://openrouteservice.org/dev/#/api-docs/geocode) to see 
```

koebi commented on docs/api-reference/endpoints/geocoder/structured-geocoding-query.md:
```diff
@@ -1,7 +1,7 @@
 # Structured Geocoding Query
```
This should be a sub-page of geocoder, no? Same as for Geocoding Response…

koebi commented on docs/api-reference/endpoints/poi-service.md:
```diff
@@ -0,0 +1,280 @@
+|:------------------------------------------------------------|:-----:|
+| [accommodation](#accommodation-100)                         |  100  |
+| [animals](#animals-120)                                     |  120  |
+| [arts_and_culture](#arts_and_culture-130)                   |  130  |
```
```suggestion
| [arts_and_culture](#arts-and-culture-130)                   |  130  |
```

koebi commented on docs/api-reference/endpoints/poi-service.md:
```diff
@@ -0,0 +1,280 @@
+| [financial](#financial-190)                                 |  190  |
+| [healthcare](#healthcare-200)                               |  200  |
+| [historic](#historic-220)                                   |  220  |
+| [leisure_and_entertainment](#leisure_and_entertainment-260) |  260  |
```
```suggestion
| [leisure_and_entertainment](#leisure-and-entertainment-260) |  260  |
```

koebi commented on docs/api-reference/endpoints/poi-service.md:
```diff
@@ -0,0 +1,280 @@
+| [historic](#historic-220)                                   |  220  |
+| [leisure_and_entertainment](#leisure_and_entertainment-260) |  260  |
+| [natural](#natural-330)                                     |  330  |
+| [public_places](#public_places-360)                         |  360  |
```
```suggestion
| [public_places](#public-places-360)                         |  360  |
```

koebi commented on docs/api-reference/endpoints/poi-service.md:
```diff
@@ -0,0 +1,280 @@
+| travel_agency |  626  |
+| viewpoint     |  627  |
+
+:::
```
Superfluous lines

koebi commented on docs/api-reference/endpoints/poi-service.md:
```diff
@@ -0,0 +1,280 @@
+| bar            |  561  |
+| bbq            |  562  |
+| biergarten     |  563  |
+| caf?           |  564  |
```
```suggestion
| café           |  564  |
```

koebi commented on docs/api-reference/error-codes.md:
```diff
@@ -84,19 +94,22 @@ EndPoints.
+|    6020    | Maximum number of visited nodes exceeded.          |
+|    6099    | Unknown internal error.                            |
+
+### Export API
```
There should be a comment down here to also keep this in sync, no?

koebi commented on docs/run-instance/index.md:
````diff
@@ -0,0 +1,21 @@
+```
+
+This will pull the latest nightly build of openrouteservice from dockerhub and start it up using an example setup and the provided test OSM file for Heidelberg/Germany and surrounding area.
+You can then modify the configuration and source file settings to match your needs. For more details, check the [Running in a container](running-in-container) section.
````
This should link to "Running with docker"?

koebi commented on docs/run-instance/system-requirements.md:
```diff
@@ -1,26 +1,28 @@
 # System Requirements
 
-When running openrouteservice, by far the biggest consideration that needs to be made is the amount of RAM that is available. Smaller areas such as Baden-Württemberg in Germany can run happily on a mid range system, but for larger areas you need more RAM. For example, in the openrouteservice infrastructure, each profile for the planet is running on a 128GB machine. Anything less than that and you ...
+When running openrouteservice, by far the biggest consideration that needs to be made is the amount of RAM that is available. Smaller areas such as Baden-Württemberg in Germany can run happily on a mid range system, but for larger areas you need more RAM. For example, in the openrouteservice public API infrastructure, each profile for the planet is running on a 128GB machine. Anything less than t ...
```
Not sure if this is up to date anymore :D

takb replied:
Yeah who cares :grin:

koebi commented on docs/run-instance/installation/index.md:
```diff
@@ -0,0 +1,9 @@
+* [Running JAR / WAR](running-jar-war)
+* [Building from Source](building-from-source)
+
+For more information on configuring openrouteservice for your specific needs, also see the [configuration](configuration/) documentation.
```
Link isn't working.

takb replied:
fixed.

koebi commented on docs/run-instance/installation/running-with-docker.md:
```diff
@@ -0,0 +1,242 @@
+
+## Checking
+
+By default the service status can be queries via the health endpoint.
```
This could link to the health endpoint site.

koebi commented on docs/run-instance/installation/running-with-docker.md:
````diff
@@ -0,0 +1,242 @@
+# {"status":"ready"}
+```
+
+When the service is ready, you will be able to request the status endpoint for further information on the running services.
````
This could link to the status endpoint page.

koebi commented on docs/run-instance/installation/running-with-docker.md:
````diff
@@ -0,0 +1,242 @@
+# Native memory allocation (mmap) failed to map 16384 bytes for committing reserved memory.
+# An error report file with more information is saved as:
+# /ors-core/hs_err_pid128.log
+```
````
Duplicate of same section in `System Requirements`?

koebi commented on docs/run-instance/installation/building-from-source.md:
````diff
@@ -0,0 +1,153 @@
+    mvn package
+   ```
+
+3. Change the `ors-engine/pom.xml`:
````
This section would also work great with the addition/deletion feature that is used in the docker-compose changing section :)

koebi commented on docs/run-instance/installation/building-from-source.md:
````diff
@@ -0,0 +1,153 @@
+   </dependency>
+   ```
+
+4. Test your new functionality and run all tests after rebasing your feature branch with the latest `development` branch. Adjust tests if necessary.
````
…the development branch of the ors itself?
Is there one called `development`?

takb replied:
this was outdated.

koebi commented on docs/run-instance/installation/building-from-source.md:
```diff
@@ -0,0 +1,153 @@
+
+5. If successful, create a PR for both [openrouteservice](https://github.com/GIScience/openrouteservice/pulls) and [GraphHopper](https://github.com/GIScience/graphhopper/pulls) against `master` and `ors_4.0` branches, respectively.
+
+**Note that in the above example, the 4.x version of GH is being used - you should adapt according to your specific version. To know which one to use, check the [ors-engine module pom file](https://github.com/GIScience/openrouteservice/ors-engine/pom.xml) and see what version is being used for the `com.github.GIScience.graphhopper` dependencies.**
```
Link isn't working.

koebi commented on docs/run-instance/configuration/index.md:
```diff
@@ -0,0 +1,32 @@
+Default settings for all configurable parameters are defined in
+[`application.yml`](https://github.com/GIScience/openrouteservice/blob/master/ors-api/src/main/resources/application.yml).
+
+If you have cloned the repository, you will find these files at `ors-api/ors-config.yml` and `ors-api/src/main/resources/application.yml` respectively. 
```
This section is a bit confusing, since it feels like there's two config files that are being read, while in fact IIUC the `application.yml` is just the "everything here as an example" file, while `ors-config.yml` is the one that should get used?

koebi commented on docs/run-instance/configuration/endpoints-and-limits.md:
```diff
@@ -0,0 +1,95 @@
+| [ors.endpoints.routing](#orsendpointsrouting)     | object | Settings for routing endpoint                                                           |
+| [ors.endpoints.isochrone](#orsendpointsisochrone) | object | Settings for the isochrones endpoint                                                    |
+| [ors.endpoints.matrix](#orsendpointsmatrix)       | object | Settings for the matrix endpoint                                                        |
+| [ors.endpoints.snap](#orsendpointssnap)           | object | Settings for the snapping endpoint                                                      |
```
Links need `-` where `.` is in key.

koebi commented on docs/run-instance/configuration/endpoints-and-limits.md:
```diff
@@ -0,0 +1,95 @@
+| attribution         | string  | Provider attribution                        |
+| provider_name       | string  |                                             |
+| provider_parameters | object  | A list of provider configuration parameters |
+| property_mapping    | object  |                                             |
```
There is two descriptions missing, and if they should be objects, there should be more explanation somewhere? Maybe make this an issue?

koebi replied:
No examples given

koebi commented on docs/run-instance/configuration/endpoints-and-limits.md:
```diff
@@ -0,0 +1,95 @@
+
+| key      | type   | description                     | example value              |
+|----------|--------|---------------------------------|----------------------------|
+| profiles | object | List of profile names           | `driving-car, driving-hgv` |
```
type is a List

koebi commented on docs/run-instance/configuration/endpoints-and-limits.md:
```diff
@@ -0,0 +1,95 @@
+| maximum_locations              | number  | Maximum number of locations per request                                                            | `2`                                                |
+| allow_compute_area             | boolean | Toggles area computation on and off                                                                | `true`                                             |
+| maximum_intervals              | number  | Maximum number of intervals/isochrones computed for each location                                  | `1`                                                |
+| fastisochrones                 | object  | Range limits for [fastisochrones](#fastisochrone-properties)                                       | `2000`                                             |
```
Type is object, example is a number

koebi commented on docs/run-instance/configuration/endpoints-and-limits.md:
```diff
@@ -0,0 +1,95 @@
+| maximum_range_distance_default | number  | Maximum default range in metres across all profiles; can be overridden in `maximum_range_distance` | `50000`                                            |
+| maximum_range_distance         | object  | List of [maximum range properties](#maximum-range-properties)                                      |                                                    |
+| maximum_range_time_default     | number  | Maximum default range in seconds across all profiles; can be overridden in `maximum_range_time`    | `18000`                                            |
+| maximum_range_time             | object  | List of [maximum range properties](#maximum-range-properties)                                      |                                                    |
```
type is object, description says list

koebi commented on docs/run-instance/configuration/engine.md:
```diff
@@ -0,0 +1,26 @@
+|-----------------------------|---------|-------------------------------------------------------------------------------------------------------------------------------------------|---------------|
+| ors.engine.init_threads     | number  | The number of threads used to initialize (build/load) graphs. Higher numbers requires more RAM                                            | `1`           |
+| ors.engine.preparation_mode | boolean | If set, graphs and preparations will be build, but the application will shut down immediately afterwards without starting up any services | `false`       |
+| ors.engine.source_file      | string  | The OSM file to be used, supported formats are `.osm`, `.osm.gz`, `.osm.zip` and `.pbf`                                                   |               |
```
No example given

koebi commented on docs/run-instance/configuration/engine.md:
```diff
@@ -0,0 +1,26 @@
+| cache_clear  | boolean | Keep elevation data once it has been downloaded                                                                                                                                                                   | `false`             |
+| provider     | string  | The name of an elevation provider. Possible values are `multi`, `cgiar` or `srtm`                                                                                                                                 | `multi`             |
+| cache_path   | string  | The path to a directory in which SRTM tiles will be stored                                                                                                                                                        | `./elevation_cache` |
+| data_access  | string  |                                                                                                                                                                                                                   | `MMAP`              |
```
No description given

koebi commented on docs/run-instance/configuration/profiles.md:
```diff
@@ -0,0 +1,43 @@
+| traffic                             | boolean | Use traffic data if available                                                                                                                                                                                                                                                                                                                                 ...
+| interpolate_bridges_and_tunnels     | boolean | Toggle elevation interpolation of bridges and tunnels on and off                                                                                                                                                                                                                                                                                              ...
+| instructions                        | boolean | Specifies whether way names will be stored during the import or not                                                                                                                                                                                                                                                                                           ...
+| optimize                            | boolean |                                                                                                                                                                                                                                                                                                                                                               ...
```
No description given

aoles replied:
I'm not sure whether it is desired to have this parameter, actually. Anyone?

sfendrich replied:
It is about optimizing the sort order when contracting nodes for CH. The operation is expensive but yields a better contraction hierarchy.

koebi commented on docs/run-instance/configuration/extras.md:
```diff
@@ -0,0 +1,20 @@
+# Additional route metadata
```
This page is linked to from "External Storages" in `profiles.md`.
This could be synced, and also synced with the sidebar.

koebi commented on docs/run-instance/configuration/extras.md:
```diff
@@ -0,0 +1,20 @@
+| WayCategory            | object | Returns the way category in the route response, Compatible for any profile type                                                                                                                                                                                                                                                                                             ...
+| WaySurfaceType         | object | Returns the way surface in the route response, Compatible for any profile type                                                                                                                                                                                                                                                                                              ...
+| HillIndex              | object | Returns the ascent/descent in the route response, Compatible for any profile type                                                                                                                                                                                                                                                                                           ...
+| TrailDifficulty        | object | Returns the trail difficulty in the route response, Compatible for profile-hiking                                                                                                                                                                                                                                                                                           ...
```
Is this meant as "If present, …"? Is this always enabled? Type `object` and default `{}` is a bit weird together…

TheGreatRefrigerator replied:
It seems to be a confusing implementation...
`"WayCategory": {}` seems to build the index, which would be 
`WayCategroy:` in yaml.

Probably there are optional options that can be set?

TheGreatRefrigerator replied:
Example from application.yml
```
        ext_storages:
          WayCategory:
          WaySurfaceType:
          Wheelchair:
            KerbsOnCrossings: true
          OsmId:
```

koebi commented on docs/run-instance/configuration/extras.md:
```diff
@@ -0,0 +1,20 @@
+| HillIndex              | object | Returns the ascent/descent in the route response, Compatible for any profile type                                                                                                                                                                                                                                                                                           ...
+| TrailDifficulty        | object | Returns the trail difficulty in the route response, Compatible for profile-hiking                                                                                                                                                                                                                                                                                           ...
+| RoadAccessRestrictions | object | RoadAccessRestrictions are where roads are restricted to certain vehicles to certain circumstances, e.g. access=destination. The use_for_warnings parameter tells the ors that this storage can be used for generating warning messages in the route response. For RoadAccessRestrictions, this means that whenever a route goes over a way which has some restrictions, a  ...
+| Wheelchair             | object | Compatible for wheelchair                                                                                                                                                                                                                                                                                                                                                   ...
```
How do I know what to write in here?

aoles replied:
I'm not sure either, do you? I just carried over the original content.

koebi commented on docs/run-instance/configuration/extras.md:
```diff
@@ -0,0 +1,20 @@
+| TrailDifficulty        | object | Returns the trail difficulty in the route response, Compatible for profile-hiking                                                                                                                                                                                                                                                                                           ...
+| RoadAccessRestrictions | object | RoadAccessRestrictions are where roads are restricted to certain vehicles to certain circumstances, e.g. access=destination. The use_for_warnings parameter tells the ors that this storage can be used for generating warning messages in the route response. For RoadAccessRestrictions, this means that whenever a route goes over a way which has some restrictions, a  ...
+| Wheelchair             | object | Compatible for wheelchair                                                                                                                                                                                                                                                                                                                                                   ...
+| OsmId                  | object | Returns the OsmId of the way, Compatible for wheelchair                                                                                                                                                                                                                                                                                                                     ...
```
As above

koebi commented on docs/run-instance/configuration/extras.md:
```diff
@@ -0,0 +1,20 @@
+| OsmId                  | object | Returns the OsmId of the way, Compatible for wheelchair                                                                                                                                                                                                                                                                                                                     ...
+| Borders                | object | Borders allows the restriction of routes to not cross country borders, compatible for any profile type                                                                                                                                                                                                                                                                      ...
+
+## Borders storage
```
Do we document border generation stuff anywhere?

aoles replied:
Nope 🙃  As this is subject to change soon, I don't think it makes sense to invest time to do it now.

koebi commented on docs/run-instance/configuration/logging.md:
```diff
@@ -0,0 +1,32 @@
+
+## Configure loglevel
+
+You can configure the loglevel for your local instance in your custom YAML config file. For example, to enable debug logging, use the following configuration.
```
Are the two examples the only available log levels?

koebi commented on docs/run-instance/configuration/logging.md:
```diff
@@ -0,0 +1,32 @@
+
+## Configure logfile
+
+The ors log file is kept by default as `logs/ors.log` relative to the runtime working directory, and is rotated daily with archived files kept as `ors.yyyy-MM-dd.log.gz`.  
```
Is logrotation implemented in ors?? That is weird, as that is not sth. the ors should be concerned with?

koebi commented on docs/run-instance/configuration/logging.md:
```diff
@@ -0,0 +1,32 @@
+## Configure logfile
+
+The ors log file is kept by default as `logs/ors.log` relative to the runtime working directory, and is rotated daily with archived files kept as `ors.yyyy-MM-dd.log.gz`.  
+You can change the log file location by setting an environment variable `ORS_LOG_LOCATION`, e.g.:
```
Can this not be configured via the config file?

takb replied:
No.

koebi commented on docs/run-instance/configuration/preparation.md:
```diff
@@ -0,0 +1,42 @@
+# Preparation settings
```
This doesn't show up in the sidebar

koebi commented on docs/run-instance/configuration/preparation.md:
```diff
@@ -0,0 +1,42 @@
+| key                      | type   | description                                                       | default value |
+|--------------------------|--------|-------------------------------------------------------------------|---------------|
+| min_network_size         | number | Minimum size (number of edges) of an independent routing subgraph | `200`         |
+| min_one_way_network_size | number |                                                                   | `200`         |
```
Description missing

sfendrich replied:
Removed as unsupported since GH1.0

koebi commented on docs/run-instance/configuration/preparation.md:
```diff
@@ -0,0 +1,42 @@
+|--------------------------|--------|-------------------------------------------------------------------|---------------|
+| min_network_size         | number | Minimum size (number of edges) of an independent routing subgraph | `200`         |
+| min_one_way_network_size | number |                                                                   | `200`         |
+| methods                  | object | [methods properties](#preparationmethods)                         |               |
```
```suggestion
| methods                  | object | [methods properties](#preparation-methods)                         |               |
```

koebi commented on docs/run-instance/configuration/preparation.md:
```diff
@@ -0,0 +1,42 @@
+
+| key  | type   | description                                            | example value                   |
+|------|--------|--------------------------------------------------------|---------------------------------|
+| ch   | object | Settings for preprocessing contraction hierarchies     | [ch](#preparationmethodsch)     |
```
```suggestion
| ch   | object | Settings for preprocessing contraction hierarchies     | [ch](#preparation-methods-ch)     |
```

koebi commented on docs/run-instance/configuration/preparation.md:
```diff
@@ -0,0 +1,42 @@
+| key  | type   | description                                            | example value                   |
+|------|--------|--------------------------------------------------------|---------------------------------|
+| ch   | object | Settings for preprocessing contraction hierarchies     | [ch](#preparationmethodsch)     |
+| lm   | object | Settings for preprocessing A* with landmarks           | [lm](#preparationmethodslm)     |
```
```suggestion
| lm   | object | Settings for preprocessing A* with landmarks           | [lm](#preparation-methods-lm)     |
```

koebi commented on docs/run-instance/configuration/preparation.md:
```diff
@@ -0,0 +1,42 @@
+|------|--------|--------------------------------------------------------|---------------------------------|
+| ch   | object | Settings for preprocessing contraction hierarchies     | [ch](#preparationmethodsch)     |
+| lm   | object | Settings for preprocessing A* with landmarks           | [lm](#preparationmethodslm)     |
+| core | object | Settings for preprocessing core routing with landmarks | [core](#preparationmethodscore) |
```
```suggestion
| core | object | Settings for preprocessing core routing with landmarks | [core](#preparation-methods-core) |
```

koebi commented on docs/run-instance/configuration/preparation.md:
```diff
@@ -0,0 +1,42 @@
+|------------|---------|----------------------------------------------------------|-------------------------|
+| enabled    | boolean | Enables or disables the routing algorithm                | `true`                  |
+| threads    | number  | Number of parallel threads for computing the preparation | `1`                     |
+| weightings | string  | Comma-separated list of weightings                       | `recommended,shortests` |
```
```suggestion
| weightings | string  | Comma-separated list of weightings                       | `recommended,shortest` |
```

koebi commented on docs/run-instance/configuration/preparation.md:
```diff
@@ -0,0 +1,42 @@
+| enabled    | boolean | Enables or disables the routing algorithm                                                                                 | `true`                                                      |
+| threads    | number  | Number of parallel threads for computing the preparation                                                                  | `1`                                                         |
+| weightings | string  | Comma-separated list of weightings                                                                                        | `recommended,shortest`                                      |
+| landmarks  | number  | Total number of precomputed landmarks, the subset used during the query is set in `execution.methods.lm.active_landmarks` | `32`                                                        |
```
```suggestion
| landmarks  | number  | Total number of precomputed landmarks, the subset used during the query is set in `execution.methods.core.active_landmarks` | `32`                                                        |
```
… I am confused :D

koebi commented on docs/run-instance/configuration/execution.md:
```diff
@@ -0,0 +1,33 @@
+# Execution settings
```
This is not linked in the sidebar

koebi commented on docs/run-instance/configuration/execution.md:
```diff
@@ -0,0 +1,33 @@
+
+| key     | type   | description | example value                |
+|---------|--------|-------------|------------------------------| 
+| methods | object |             | [methods](#executionmethods) |
```
```suggestion
| methods | object |             | [methods](#execution-methods) |
```

koebi commented on docs/run-instance/configuration/execution.md:
```diff
@@ -0,0 +1,33 @@
+|------|--------|-------------------------------------------------------|-------------------------------| 
+| ch   | object | Settings for using contraction hierarchies in routing | [ch](#executionmethodsch)     |
+| lm   | object | Settings for using landmarks in routing               | [lm](#executionmethodslm)     |
+| core | object | Settings for using landmarks in routing               | [core](#executionmethodscore) |
```
```suggestion
| ch   | object | Settings for using contraction hierarchies in routing | [ch](#execution-methods-ch)     |
| lm   | object | Settings for using landmarks in routing               | [lm](#execution-methods-lm)     |
| core | object | Settings for using landmarks in routing               | [core](#execution-methods-core) |
```

koebi commented on docs/run-instance/configuration/execution.md:
```diff
@@ -0,0 +1,33 @@
+
+| key               | type    | description | example value |
+|-------------------|---------|-------------|---------------| 
+| disabling_allowed | boolean |             | `true`        |
```
Description missing

sfendrich replied:
Removed as not supportet since GH3

koebi commented on docs/run-instance/configuration/execution.md:
```diff
@@ -0,0 +1,33 @@
+
+| key               | type    | description                                      | default value |
+|-------------------|---------|--------------------------------------------------|---------------| 
+| disabling_allowed | boolean |                                                  | `true`        |
```
Description missing.

sfendrich replied:
Removed as not supportet since GH3

koebi commented on docs/run-instance/configuration/execution.md:
```diff
@@ -0,0 +1,33 @@
+
+| key               | type    | description                                      | example value |
+|-------------------|---------|--------------------------------------------------|---------------| 
+| disabling_allowed | boolean |                                                  | `true`        |
```
Description missing

sfendrich replied:
Removed as not supportet since GH3

koebi commented on docs/api-reference/index.md:
```diff
@@ -0,0 +1,28 @@
+  Complete and concise technical documentation of our live API. 
+  It can also be used to send sample requests to both, our live API or an instance of openrouteservice running on your local machine. 
+  Use the form "API Server" on the top left to select and edit "Development Server" if you want to send requests to your local instance:
+  ![](../public/playground-select-server.png)
```
Files in the public directory are served at the root path.
Instead of /public/playground-select-server.png, use /playground-select-server.png.

koebi commented on docs/api-reference/index.md:
```diff
@@ -0,0 +1,28 @@
+
+  The responses of your sample requests are visualized on a small map on the bottom right, 
+  but can also be shown as json, table or downloaded as file:
+  ![](../public/playground-map.png)
```
Files in the public directory are served at the root path.
Instead of /public/playground-map.png, use /playground-map.png.

koebi commented on docs/api-reference/index.md:
```diff
@@ -0,0 +1,28 @@
+  You can then navigate to the swagger-ui hosted by your local instance in your browser: [http://localhost:8082/ors/swagger-ui/index.html](http://localhost:8082/ors/swagger-ui/index.html)
+  The port may be different on your local environment.
+  **The swagger-ui is not available on the production ORS instances hosted by HeiGIT.**
+  ![](../public/swagger-ui.png)
```
Files in the public directory are served at the root path.
Instead of /public/swagger-ui.png, use /swagger-ui.png.

koebi commented on docs/run-instance/configuration/json.md:
```diff
@@ -1,109 +1,10 @@
-# Configuration
+# Configuration via `ors-config.json`
```
There's a bunch of issues in here, links are same as in the other files and a lot of other things as well.
Not sure if it's worth fixing…

TheGreatRefrigerator replied:
Fixed links and table format

koebi commented on docs/contributing/contributing-translations.md:
```diff
@@ -3,7 +3,7 @@ If you notice any thing wrong with translations, or you want to add a new langua
 As a guide, for each instruction there are two formats - one where there is a named place (i.e. 47 Berlinerstraße) and one without. It is important to keep the `{way_name}` tag in the text, but it should be moved to the correct location.
```
https://github.com/GIScience/openrouteservice/blob/master/ors-engine/src/main/resources/resources/locales/ors_en-US.resources is the corect link to `ors-en-US.resources`

takb replied:
addressed

koebi commented on docs/contributing/contributing-translations.md:
```diff
@@ -3,7 +3,7 @@ If you notice any thing wrong with translations, or you want to add a new langua
 To show openrouteservice what language is being used, you should alter the name of the file to include the ISO identifier (i.e. for Austrian German the filename would be `ors_de_AT.resources`).
 
-Since the update to the v2 API, we make use of the Spring framework. Because of that, you also need to make a slight modification to a java file that declares what parameters can be passed through to the API. This file is located in [`openrouteservice/src/main/java/heigit/ors/api/requests/common/APIEnums.java`](https://github.com/GIScience/openrouteservice/blob/master/ors-engine/src/main/java/org ...
+Since the update to the v2 API, we make use of the Spring framework. Because of that, you also need to make a slight modification to a java file that declares what parameters can be passed through to the API. This file is located in [`ors-engine/src/main/java/org/heigit/ors/routing/APIEnums.java`](https://github.com/GIScience/openrouteservice/blob/master/ors-engine/src/main/java/org/heigit/ors/ro ...
```
```suggestion
Since the update to the v2 API, we make use of the Spring framework. Because of that, you also need to make a slight modification to a java file that declares what parameters can be passed through to the API. This file is located in [`ors-engine/src/main/java/org/heigit/ors/routing/APIEnums.java`](https://github.com/GIScience/openrouteservice/blob/master/ors-api/src/main/java/org/heigit/ors/api/APIEnums.java) and the section that needs updating is the `public enum Languages` enum. If (for example) you want to add Polish to the languages, you would also need to add here something like `PL("pl")` to the list where the part in the `("...")` is what would be passed in to the API to select that language. So ultimately you would end up with:
```

koebi commented on docs/technical-details/travel-speeds/index.md:
```diff
@@ -13,38 +13,38 @@ adjusting them for different
 [surfaces](https://wiki.openstreetmap.org/wiki/Key:surface) of the road.
```
Again, grades might not be the best here - see suggestions above.

koebi commented on ors-api/src/main/java/org/heigit/ors/api/EngineProperties.java:
Not part of the docs overhaul

koebi commented on ors-api/src/main/resources/log4j-tomcat.properties:
Not part of the docs overhaul

TheGreatRefrigerator replied:
Reverted

koebi commented on docs/technical-details/tag-filtering.md:
Are we sure that this page is currently accurate?

TheGreatRefrigerator replied:
we are now more or less, still waiting for a summary of @aoles for the general conditionals used by graphhopper:

aoles replied:
Done, see below.

koebi commented on docs/frequently-asked-questions.md:
Link to configuration not working.

koebi replied:
Link to system requirements also not working

TheGreatRefrigerator replied:
Done

koebi commented on docker-entrypoint.sh:
```diff
@@ -3,7 +3,7 @@
 
 if [[ -d /ors-core ]] || [[ -d /ors-conf ]]; then
-  echo "You're mounting old paths. Remove them and migrate to the new docker setup: https://giscience.github.io/openrouteservice/installation/Running-with-Docker.html"
+  echo "You're mounting old paths. Remove them and migrate to the new docker setup: https://giscience.github.io/openrouteservice/run-instance/installation/running-in-container.html"
```
```suggestion
  echo "You're mounting old paths. Remove them and migrate to the new docker setup: https://giscience.github.io/openrouteservice/run-instance/installation/running-with-docker"
```

koebi reviewed (changes requested):
Hey,

great work so far! This is definitely moving in the right direction, and I do like the structure you came up with :)
Is there a spell checker that we can run on this easily/automatedly? Haven't seen any major typos, but might be worth setting this up.

I have only skimmed the installation part in regards to the content.

Do we have an explanation of CH/LM/Core anywhere, or can we maybe link to some article(s)?

Is the order in which config options are read (ENV > ORS_CONFIG > appl.yml) documented?

There's an overarching issue in a few places:
`The language 'jsonpath' is not loaded, falling back to 'txt' for syntax highlighting.`

Are the values of the track- and waytype speed tables aligned with the speed value files?

Do we link to the documentation from github? Do we link to it from openrouteservice.org or from the API playground?

takb committed:
docs: link to datatracker for GeoJSON structure

it has better visualisation

Co-authored-by: Jakob Schnell <Jakob.Schnell@heigit.org>

aoles commented on docs/run-instance/installation/building-from-source.md:
```diff
@@ -0,0 +1,153 @@
+After running this command, you will find the artefact at `ors-api/target/ors.war`. 
+
+After you have packaged openrouteservice, there are two options for running it.
+One is to run the `mvn spring-boot:run` command which triggers a spring-boot native
```
I've tried following the instructions but run into an error upon `mvn spring-boot:run`. Setting the env variable `export ORS_CONFIG_LOCATION=ors-config.yml` solved the issue for me.
```
24 Jan 10:35:41 ERROR                                             ORS-Init [ o.h.o.r.RoutingProfileManager            ]   
24 Jan 10:35:41 ERROR                                             ORS-Init [ o.h.o.r.RoutingProfileManager            ]   No profiles configured. Exiting.
24 Jan 10:35:41 ERROR                                             ORS-Init [ o.h.o.r.RoutingProfileManager            ]   
```

aoles reviewed (dismissed):
Instructions on running ORS form the command line need to be updated.

takb committed:
docs(config): add jsonpath code block option

Supposedly this should fix said warnings according to https://github.com/vuejs/vitepress/issues/3259,
though locally it didn't seem to help.
Need to investigate further when we have time.

takb committed:
docs: fix country table

jhaeu committed:
docs(routing-options): update page content

- remove url encoding part, which is not needed for passing options in POST request
- link vehicle_type to tag filtering
- remove routing restriction option 'gradient' as support was removed
- use jsonpath for options parameter headings

docs(routing-options): adjust examples

- check for validity and fix
- remove pretty code block if minified is readable
- make minified examples code blocks for quick copy

Co-authored-by: Amandus <amandus.butzer@heigit.org>

TheGreatRefrigerator commented on docs/run-instance/installation/building-from-source.md:
```diff
@@ -0,0 +1,150 @@
+  1. Set up your IDE project and import `openrouteservice`
+     modules as Maven model.
+     For IntelliJ Idea, have a look at [these instructions](/contributing/opening-project-in-intellij).
+
```
When using the Intellij generated "spring boot Application" run configuration. This doesn't work out of the box for me. I needed to set the Working directory to <repo_root>/ors-api for the default config paths to work.

The current paths in ors-config.yml also seem to be done for the docker setup only. As i now get ors-core folders spawned, wherever the current working directory is set (root or ors-api folder), which are also not ignored?

TheGreatRefrigerator replied:
Things that should eventually work out of the box:

Intellij run configs for:

- autogenerated Spring Boot task -> RECOMMENDED
- maven spring-boot:run task (on openrouteservice or ors-api folder?)
    (maybe provide some info on what you normally run when developing ORS

Command line:
mvn spring-boot:run

docker compose up -d
docker run ...

java -jar ors.jar


Tests should work both with
- ors-api/src/tests -> right-click run tests
- mvn test
- mvn -Papitests verify

TheGreatRefrigerator replied:
jar / war & docker setup will be tackled outside this PR

TheGreatRefrigerator committed:
docs: adjust to default branch rename (master->main)

Hydeo commented on docs/run-instance/installation/running-with-docker.md:
```diff
@@ -0,0 +1,226 @@
+
+1. Pull the openrouteservice Docker image from Dockerhub and start a container named `ors-app`
+2. Launch the openrouteservice service on port `8080` within a tomcat running in that container, available at the address `http://localhost:8080/ors`.
+3. A local `./docker` folder containing the files used and produced by openrouteservice for easy access. Most relevant is `./docker/conf/ors-config.yml` controlling ORS behaviour, and the test OSM data file  `/home/ors/ors-core/data/osm_file.pbf` of Heidelberg and surroundings.
```
`A local` -> `Create a local`

koebi committed:
chore: bump vitepress version

jhaeu committed:
docs(endpoints): revise error codes

TheGreatRefrigerator committed:
docs(config): ignore dead links to localhost

MichaelsJP commented on ors-api/ors-config.yml:
This file was moved to the root folder and shouldn't be in ors-api/ any more. Maybe it makes sense to add the old location to the .gitignore . Else we'll prob have more accidental commits once in a while.

takb replied:
done

takb committed:
chore: add old ors-config.yml location to .gitignore

TheGreatRefrigerator committed:
docs(tag-filtering): revise and fix profiles

- add info for conditional access restrictions
- introduce default case per profile
- remove false info of ways being always rejected

Co-authored-by: Andrzej K. Oleś <andrzej.oles@gmail.com>

koebi committed:
docs: fix typos

TheGreatRefrigerator commented on .rpm-packaging/ors-config.json:
```diff
@@ -152,14 +149,11 @@
               "execution": {
                 "methods": {
                   "ch": {
```
empty object?

TheGreatRefrigerator commented on .rpm-packaging/ors-config.json:
```diff
@@ -202,10 +195,8 @@
               "execution": {
                 "methods": {
                   "ch": {
```
empty object -> remove

TheGreatRefrigerator commented on docs/api-reference/endpoints/directions/extra-info/index.md:
````diff
@@ -0,0 +1,146 @@
+$.routes[*].extras
+```
+
+of type map. Each requested extra_info is represented as a nested map entry with the requested value as key, e.g. `steepness`.
````
i would use JSON terms:
map -> object 

... in the extras object at:
``
``

Each .. .. nested object ....`extras` object

TheGreatRefrigerator commented on docs/api-reference/endpoints/directions/extra-info/index.md:
````diff
@@ -0,0 +1,146 @@
+}
+```
+
+The values of each entry have the same structure: Maps with the entries `values` and `summary`:
````
as above JSON terms:
Objects with the keys ... 
+ md link to value and summary headings

TheGreatRefrigerator commented on docs/api-reference/endpoints/directions/extra-info/index.md:
```diff
@@ -0,0 +1,146 @@
+ 
+### `summary`
+
+A list of maps with the entries `value`, `distance`, `amount` summarizing the 
```
maps -> objects, entries -> keys/properties

TheGreatRefrigerator commented on docs/api-reference/endpoints/directions/extra-info/road-access-restrictions.md:
```diff
@@ -0,0 +1,22 @@
+|  16   |              Private              |
+|  32   |            Permissive             |
+
+[//]: # (keep in sync with AccessRestrictionType.class)
```
Do we not need the same in the code to sync as comment? (for all of these keep in sync things)

jhaeu replied:
good idea, added bi-directional hints

TheGreatRefrigerator commented on docs/api-reference/endpoints/directions/requests-and-return-types.md:
```diff
@@ -0,0 +1,33 @@
+
+### JSON 
+
+The **JSON** return type is an optimized proprietary openrouteservice format and best suited for further processing.
```
so JSON is an optimized proprietary openrouteservice format ? 😂 
Probably rephrase this

sfendrich replied:
```suggestion
The **JSON** return type is best suited for further processing.
```

TheGreatRefrigerator commented on docs/api-reference/endpoints/directions/routing-options.md:
```diff
@@ -0,0 +1,264 @@
+
+With the request body parameter `options`, advanced routing options can be specified for a directions request.
+
+The `options` parameter is a JSON map, multiple of the here mentioned parameters can be set. 
```
map -> object throughout the whole file

TheGreatRefrigerator commented on docs/api-reference/endpoints/isochrones/index.md:
```diff
@@ -0,0 +1,14 @@
+
+The Isochrone Service supports time and distance analyses for one single or multiple locations.
+It is possible to specify the isochrone interval or provide multiple exact isochrone range values.
+This service allows the same range of profile options as the /directions endpoint,
```
/directions -> code format

koebi replied:
Done

TheGreatRefrigerator commented on docs/api-reference/endpoints/matrix/index.md:
````diff
@@ -0,0 +1,60 @@
+```json
+{
+  "locations": [
+    [8.660445213317873,49.41571970315377],
````
presicion 7 digits max

koebi replied:
Done

TheGreatRefrigerator commented on docs/contributing/index.md:
```diff
@@ -1,8 +1,4 @@
-* [Updating OpenStreetMap](https://wiki.openstreetmap.org/wiki/Beginners%27_guide)
-
 When people contribute to openrouteservice, it makes [the whole
 team](https://heigit.org/heigit-team/) do a little happy dance!  There are many
```
Remove first sentence.
Put team info somewhere else if needed.

koebi replied:
Done

TheGreatRefrigerator commented on docs/contributing/index.md:
```diff
@@ -12,14 +8,17 @@ information in OSM so that openrouteservice can take it into account, or even
 
-Feel free to create an [issue](https://github.com/GIScience/openrouteservice/issues)
-and label it accordingly. If your issue regards the openrouteservice web-app
+Feel free to create an [issue](https://github.com/GIScience/openrouteservice/issues) and label it accordingly. If your issue regards the openrouteservice web-app
```
probably say maps.openrouteservice.org or ors maps client to avoid confusion with ors api which is also a web app

koebi replied:
Done

TheGreatRefrigerator commented on docs/contributing/index.md:
```diff
@@ -12,14 +8,17 @@ information in OSM so that openrouteservice can take it into account, or even
-We put together a [few
-guidelines](https://github.com/GIScience/openrouteservice/blob/master/CONTRIBUTE.md)
+To help you get started, here is a document describing how to [open the code in IntelliJ](opening-project-in-intellij).  
+We also put together a few [guidelines](https://github.com/GIScience/openrouteservice/blob/main/CONTRIBUTE.md)
```
These guidelines should be checked at some point as well.

koebi replied:
I transferred this into a new issue, since it's beyond the scope of this PR

TheGreatRefrigerator commented on docs/frequently-asked-questions.md:
```diff
@@ -23,9 +23,9 @@ There are three main reasons for this problem, listed in order of most to least
    road. The maximum distance for snapping to road segments in our API is 350m.
  This can be customized for local installations via the
   `maximum_snapping_radius` and `location_index_resolution` config-parameter. See
```
both of these config parameters don't exist anymore?

TheGreatRefrigerator commented on docs/getting-started.md:
```diff
@@ -1,20 +1,27 @@
+* [Technical details](technical-details/) contains all sorts of explanation about the internal workings of the openrouteservice.
+* [FAQ](frequently-asked-questions) contains answers to frequently asked questions.
+
+For a more general overview, visit our documentation at [openrouteservice.org](https://openrouteservice.org).
```
documentation -> website

koebi replied:
Done

jhaeu committed:
docs: bi-directional hints to sync docu

TheGreatRefrigerator commented on docs/run-instance/configuration/index.md:
```diff
@@ -0,0 +1,92 @@
+The option to configure using environment variables is especially useful in contexts where you want to run openrouteservice in containers such as with [docker](../installation/running-with-docker). 
+
+Every property also corresponds to an environment variable name in *uppercase letters* and with *underscores* replacing *dots*, so e.g.
+- `ORS_ENGINE_SOURCE_FILE` replaces `ors.engine.source_file`
```
As far as i understood Julian, this is also working with dot notation?
should check

TheGreatRefrigerator replied:
I just didn't read that block properly. This is described already here.
However, the reporting for overwriting with Env variables did only work for the uppercase versions.
This is now fixed

TheGreatRefrigerator commented on docs/run-instance/configuration/ors/engine/profiles.md:
```diff
@@ -0,0 +1,150 @@
+
+## encoder_options
+
+Properties beneath `ors.engine.profiles.`\<profile>`.encoder_options`:
```
jsonpath with wildcard `.profiles.*.encoder_options` instead?

TheGreatRefrigerator replied:
also the ones below, if we do this

TheGreatRefrigerator commented on docs/run-instance/configuration/spring/server.md:
```diff
@@ -0,0 +1,8 @@
+# `server` 
+
+General server properties used by spring. The most important ones are:
```
link to other server properties, if this is not a complete lits

TheGreatRefrigerator commented on docs/run-instance/data.md:
```diff
@@ -0,0 +1,18 @@
+The data used for elevation are [SRTM](http://srtm.csi.cgiar.org/) and [GMTED](https://www.usgs.gov/coastal-changes-and-impacts/gmted2010).  
+
+## Population
+When requesting isochrones, you can also request to get population data for the isochrone areas. The data used for this is the [Global Human Settlement Layer (GHSL)](https://ghsl.jrc.ec.europa.eu/ghs_pop2023.php) from the European Commission. openrouteservice uses the 2020 data at a resolution of 100m.
```
nit:
data 2020
dataset 2023

TheGreatRefrigerator commented on docs/run-instance/data.md:
```diff
@@ -0,0 +1,18 @@
+Data relating to the avoid borders features is derived from administrative boundaries features in OpenStreetMap. Information about open borders is obtained from [Wikipedia](https://en.wikipedia.org/wiki/Open_border).
+
+## Green & Quiet
+The data used to identify green and quiet routes were derived from research projects in the GIScience research group at Heidelberg University. More information about these can be found on the GIScience news blog [here](http://k1z.blog.uni-heidelberg.de/2017/07/03/healthy-routing-prefering-green-areas-added-to-openrouteserviceorg/) and [here](http://k1z.blog.uni-heidelberg.de/2017/07/10/reducing-s ...
```
link to new url
https://giscienceblog.uni-heidelberg.de/2017/07/03/healthy-routing-prefering-green-areas-added-to-openrouteserviceorg/
https://giscienceblog.uni-heidelberg.de/2017/07/10/reducing-stress-by-avoiding-noise-with-quiet-routing-in-openrouteservice/

koebi replied:
Done

TheGreatRefrigerator commented on docs/run-instance/installation/running-jar-war.md:
```diff
@@ -0,0 +1,46 @@
+## Folders and files
+
+Openrouteservice produces output files of three types, for which the paths can be configured. The directories these paths point to need to be *writable*. The default configuration file for openrouteservice mentioned above provides the following paths:
+- `./ors-core/data/graphs`: The converted graphs for the profiles are stored here.
```
adjust default paths to ./

TheGreatRefrigerator replied:
jar/war setup outside this PR

TheGreatRefrigerator commented on docs/technical-details/tag-filtering.md:
```diff
@@ -0,0 +1,158 @@
+`Conditional` means that the tag is taken into account during the filtering process,
+but the acceptance/rejection is based on other tags and logic, such as the [`:conditional` suffix](https://wiki.openstreetmap.org/wiki/Conditional_restrictions).
+
+**All ways are initially rejected from the graph building unless criteria for acceptance has been met!**
```
This is somehow not true. @aoles wanted to write a few sentences on how graphhopper handles this.

aoles replied:
I've added a paragraph on how time-dependent restrictions are being currently handled.

TheGreatRefrigerator commented on docs/technical-details/travel-speeds/index.md:
```diff
@@ -70,7 +70,7 @@ servers as it can lead to undesirable routes.
 The travel speeds for `foot-*` profiles (walking and hiking) are set to
 5 km/h on all allowed waytypes.
-For ways with a [`sac_scale`](../extra-info/Trail-Difficulty) higher than
+For ways with a [`sac_scale`](../../api-reference/endpoints/directions/extra-info/trail-difficulty.md) higher than
```
probably use absolute paths. + mark docs folder as sources root (that's the root for vitepress) to make md links work

koebi replied:
This is an IntelliJ-related issue, no?

TheGreatRefrigerator replied:
intellij/vitepress. As `<reporoot>/docs` is defined as the root in vitepress. all links can be relative to that folder e.g.:
`[these instructions](/contributing/opening-project-in-intellij)` in `building-from-source.md`
They also work without the `.md` extension but intellij only finds the path with extension.
My question is if we should do this the vitepress or the intellij/filesystem way?

TheGreatRefrigerator replied:
https://vitepress.dev/guide/routing#linking-between-pages
suggests to not use endings but is fine with either relative or absolute (relative to vitepress base which is the docs folder).
Personal preference would be absolute, but file links definitely should work in intellij with strg + click.
This currently only works if the `docs` folder is marked as 'sources root' and the `.md` ending is put in the link

TheGreatRefrigerator commented on ors-api/ors-config.yml:
```diff
@@ -0,0 +1,196 @@
+##### openrouteservice settings file #####
```
this config file is not needed

TheGreatRefrigerator commented on ors-engine/src/main/java/org/heigit/ors/routing/graphhopper/extensions/ORSGraphHopper.java:
```diff
@@ -88,8 +88,7 @@ public class ORSGraphHopper extends GraphHopperGtfs {
 
-    private int minNetworkSize = 200;
-    private int minOneWayNetworkSize = 0;
+    private int minNetworkSize = 200;//TODO: does this parameter actually influence anything?
```
yes, if you have subnetworks in the graph below this nodecount (or edgecount) they are removed from the graph.

TheGreatRefrigerator replied:
Removed

TheGreatRefrigerator commented:
@koebi should we exclude the deprecated JSON config from search? As it shows up a lot for configuration parameter searches..

jhaeu committed:
docs: add link to mmap config in  FAQ

- remove double horizontal line

TheGreatRefrigerator committed:
docs: use absolute links for .. scope

- add some header reference to links

Note: Intellij still doesn't resolve dashed header references
properly

TheGreatRefrigerator committed:
docs: add info on resolving markdown links in IDE

TheGreatRefrigerator committed:
docs: remove config file for old GitHub pages

TheGreatRefrigerator committed:
docs: move to new structure

Co-authored-by: Jakob Schnell <Jakob.Schnell@heigit.org>
Co-authored-by: sfendrich <sascha.fendrich@uni-heidelberg.de>
Co-authored-by: Takara Baumbach <takara.baumbach@heigit.org>
Co-authored-by: Jochen Haeussler <jochen.haeussler@heigit.org>
Co-authored-by: Andrzej K. Oleś <andrzej.oles@gmail.com>

TheGreatRefrigerator committed:
docs: adjust important links to new structure

Co-authored-by: Jakob Schnell <Jakob.Schnell@heigit.org>
Co-authored-by: sfendrich <sascha.fendrich@uni-heidelberg.de>
Co-authored-by: Takara Baumbach <takara.baumbach@heigit.org>
Co-authored-by: Jochen Haeussler <jochen.haeussler@heigit.org>
Co-authored-by: Andrzej K. Oleś <andrzej.oles@gmail.com>

TheGreatRefrigerator committed:
docs(api-reference): complete section rework

- rework intro and add images
- add entry for every endpoint
- add warnings for endpoint availability
- update & correct information
- add general info on requests-and-return-types.md

- include routing-options examples
- remove unused Structured-Geocoding-Query.md

Co-authored-by: Jakob Schnell <Jakob.Schnell@heigit.org>
Co-authored-by: sfendrich <sascha.fendrich@uni-heidelberg.de>
Co-authored-by: Takara Baumbach <takara.baumbach@heigit.org>
Co-authored-by: Jochen Haeussler <jochen.haeussler@heigit.org>
Co-authored-by: Andrzej K. Oleś <andrzej.oles@gmail.com>

TheGreatRefrigerator committed:
docs(api-reference): revise extra-info section

- update info & tables
- add osm wiki link for osm way IDs
- add info on shadow and csv
- add jsonpath for where to find in response
- add 'keep in sync' hints to code counterpart
- add extra info response example

Co-authored-by: Jakob Schnell <Jakob.Schnell@heigit.org>
Co-authored-by: sfendrich <sascha.fendrich@uni-heidelberg.de>
Co-authored-by: Takara Baumbach <takara.baumbach@heigit.org>
Co-authored-by: Jochen Haeussler <jochen.haeussler@heigit.org>
Co-authored-by: Andrzej K. Oleś <andrzej.oles@gmail.com>

TheGreatRefrigerator committed:
docs(contributing): rework section

TheGreatRefrigerator committed:
docs(run-instance): complete section rework

- move to dedicated yaml config documentation following application.yml structure
- mark json config as deprecated
- update and fix information, tables & links

Co-authored-by: Jakob Schnell <Jakob.Schnell@heigit.org>
Co-authored-by: sfendrich <sascha.fendrich@uni-heidelberg.de>
Co-authored-by: Takara Baumbach <takara.baumbach@heigit.org>
Co-authored-by: Jochen Haeussler <jochen.haeussler@heigit.org>
Co-authored-by: Andrzej K. Oleś <andrzej.oles@gmail.com>

TheGreatRefrigerator committed:
docs(run-instance): update installation section

- update, correct and restructure information
- add dedicated running-jar-war.md file with draft for tomcat 10
- remove unused Advanced-Docker-Setup.md

Co-authored-by: Jakob Schnell <Jakob.Schnell@heigit.org>
Co-authored-by: sfendrich <sascha.fendrich@uni-heidelberg.de>
Co-authored-by: Takara Baumbach <takara.baumbach@heigit.org>
Co-authored-by: Jochen Haeussler <jochen.haeussler@heigit.org>
Co-authored-by: Andrzej K. Oleś <andrzej.oles@gmail.com>

TheGreatRefrigerator committed:
docs(technical-details): update section

- add index.md with general info and links
- fix some infos & links
- remove emojis from headings

Co-authored-by: Jakob Schnell <Jakob.Schnell@heigit.org>
Co-authored-by: sfendrich <sascha.fendrich@uni-heidelberg.de>
Co-authored-by: Takara Baumbach <takara.baumbach@heigit.org>
Co-authored-by: Jochen Haeussler <jochen.haeussler@heigit.org>
Co-authored-by: Andrzej K. Oleś <andrzej.oles@gmail.com>

TheGreatRefrigerator committed:
docs: remove old v1 to v2 migration instructions

TheGreatRefrigerator committed:
docs: add CHANGELOG.md entry

TheGreatRefrigerator commented:
Restrucutred to 26 commits, with a few more fixed typos an the last adjustments to tag-filtering.
See https://github.com/GIScience/openrouteservice/compare/63614f1514672bcc1f32e0451a679434119cb6dc..3336135cbdc0a8e04057a948d614289ec4853b2c for diff

jhaeu reviewed (approved):
LGTM

koebi reviewed (approved):
LGTM :)
